OpenRA.Game/Server/Server.cs

```diff
@@ -660,16 +660,26 @@ void ValidateClient(Connection newConn, string data)
 			}
 		}
 
+		byte[] CreateFrame(int client, int frame, byte[] data)
+		{
+			var ms = new MemoryStream(data.Length + 12);
+			ms.WriteArray(BitConverter.GetBytes(data.Length + 4));
+			ms.WriteArray(BitConverter.GetBytes(client));
+			ms.WriteArray(BitConverter.GetBytes(frame));
+			ms.WriteArray(data);
+			return ms.ToArray();
```

Comment:
Do not use `ToArray` when you know that `ms.Capacity == ms.Length`, use `GetBuffer`.
Also, dispose the MemoryStream.